### PR TITLE
Update of Terraform - Flatten, Eslint, refactor, config.json and README.md improvements.

### PR DESCRIPTION
**Pull request by kenskyfish**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  ],
+  parserOptions: {
+    ecmaVersion: "latest",
+  },
+  rules: {
+    quotes: ["error", "double"],
+    curly: ["error", "all"],
+    "no-console": ["warn", { allow: ["info", "warn", "error"] }],
+    "no-use-before-define": ["error", { functions: false }],
+    // "no-empty": ["error", { allowEmptyCatch: true }],
```
Comment (graddaniel) [inline .eslintrc.js]: If some rules are meant to be disobeyed, let's remove them then.

File: config.json [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      "learnUrl": "https://learn.hashicorp.com/tutorials/terraform/init"
+    }
+  ],
+  "methods": [
+    {
+      "name": "executeTerraformInit",
+      "viewName": "Init",
+      "params": [
+        {
+          "name": "workingDir",
```
Comment (graddaniel) [inline config.json]: Please revert all the `name` fields changed in this PR to the original form. I realise they look bad, but these are the params and methods identifiers, so if we change them existing pipelines will break.

That's definitely something that needs to be fixed some day, but for now it's better work with the old incorrectly named variables, than make the clients lose their settings.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+async function exeTf(mode, action, settings) {
+  const path = action.params.workingDir || settings.workingDir;
+  let args;
+  if (mode === "init") {
+    args = action.params.upgrade ? ["-upgrade"] : [];
+  } else {
+    args = parseVars(action.params.vars);
+    if (action.params.varFile) { args.push(`-var-file="${action.params.varFile}"`); }
+    if (mode === "apply" || mode === "destroy") { args.push("--auto-approve"); }
```
Comment (graddaniel) [inline app.js]: This condtion could be placed outsite the `else` statement. The fewer nestings, the easier it's to understand. Also let's group conditons of similar abstractions (`mode` comparison in this case) at the same nesting level (if possible, without breaking the logic).

**Review by graddaniel — CHANGES_REQUESTED** [review]
It's great that you've decided to separate the feature and refactoring. It's much easier to review it. Looks ok, but I have a couple of requests.

File: .eslintrc.js (hunk 1 above)
Comment (kenskyfish) [reply .eslintrc.js]: yes.

File: config.json (hunk 2 above)
Comment (kenskyfish) [reply config.json]: yes

File: app.js (hunk 3 above)
Comment (kenskyfish) [reply app.js]: yes

Comment (kenskyfish) [reply app.js]: done

File: .eslintrc.js (hunk 1 above)
Comment (kenskyfish) [reply .eslintrc.js]: done

**Review by graddaniel — APPROVED** [review]

**Pull request merged**